text/0000-stabilize-marker-freeze.md

```diff
@@ -91,46 +89,75 @@ in read-only static memory or writeable static memory, or the optimizations that
 in code that holds an immutable reference to `T`.
 
 # Semver hazard
-`Freeze` being an auto-trait, it may leak private properties of your types to semver.
-Specifically, adding a non-`Freeze` field to a type's layout is a _major_ breaking change,
-as it removes a trait implementation from it. Removing the last non-`Freeeze` field of a type's
-layout is a _minor_ breaking change, as it adds a trait implementation to that type.
+`Freeze` being an auto-trait that encodes a low level property of the types it is implemented for, 
+it is strongly advised against to rely on external types maintaining that property unless that
+contract is explicitly stated out-of-band (through documentation, for example).
+
+Conversely, authors that consider `Freeze` to be part of a type's contract should document this
+fact explicitly.
 
 ## The ZST caveat
-While `UnsafeCell<T>` is currently `!Freeze` regardless of `T`, allowing `UnsafeCell<T>: Freeze` iff `T` is
+While `UnsafeCell<T>` is currently `!Freeze` regardless of `T`, allowing `UnsafeCell<T>: Freeze` if `T` is
 a Zero-Sized-Type is currently under consideration.
 
 Therefore, the advised way to make your types `!Freeze` regardless of their actual contents is to add a 
 [`PhantomNotFreeze`](core::marker::PhantomNotFreeze) field to it.
 
+[`PhantomData<T>`](core::marker::PhantomData) only implements `Freeze` if `T` does, making it a good way
+to conditionally remove a generic type's `Freeze` auto-impl.
```

Comment:
This is currently not true (`PhantomData<T>` is always `Freeze`). And when I tried changing that there was significant fallout, though I do not remember the details.

Reply:
Arguably, this is an example of what the `Freeze` doc could/should look like _once implemented_ :p